consensus/polybft/fsm.go

```diff
@@ -604,8 +617,8 @@ func validateHeaderFields(parent *types.Header, header *types.Header) error {
 		return fmt.Errorf("mix digest is not correct")
 	}
 	// difficulty must be > 0
-	if header.Difficulty <= 0 {
-		return fmt.Errorf("difficulty should be greater than zero")
+	if header.Difficulty != 0 {
+		return fmt.Errorf("difficulty should be zero")
```

Comment:
Not sure if it is worth insisting on difficulty being set to 0 since there is a https://github.com/0xPolygon/polygon-edge/pull/831, which removes it altogether.